### PR TITLE
Improve softmax ONNX export tests

### PR DESCRIPTION
**Pull request by nzmora-nvidia**
* Use ORT CUDA backend by default.
* Add dynamically shaped input mask in test_export_softmax
* Fix test_softmax_mask_fn - use env. var `NVTE_ONNX_KVCACHE_MAX_SEQ_LEN` to control whether the test uses the default mask generation function or dynamic TRILU mask slicing.

### CODE REVIEW

File: tests/pytorch/test_onnx_export.py [hunk 1]
```diff
@@ -1129,14 +1138,14 @@ def test_export_layernorm_mlp(
...
-    (torch.float32,  False,    None),      # calls forward_torch_softmax
-    (torch.float32,  True,     None),      # calls forward_torch_softmax
-    (torch.float16,  False,    "causal"),  # calls ScaledUpperTriangMaskedSoftmax
-    (torch.float16,  True,     "padding"), # calls ScaledMaskedSoftmax
-    (torch.float16,  False,    "padding"), # calls ScaledSoftmax
-    (torch.bfloat16, False,    "causal"),  # calls ScaledUpperTriangMaskedSoftmax
-    (torch.bfloat16, True,     "padding"), # calls ScaledMaskedSoftmax
-    (torch.bfloat16, False,    "padding"), # calls ScaledSoftmax
+    (torch.float32,  True,     "padding"), # calls forward_torch_softmax (apply user mask)
+    (torch.float32,  False,    "padding"), # calls forward_torch_softmax (apply no mask)
```
Comment (ksivaman) [inline tests/pytorch/test_onnx_export.py:1142]:
```suggestion
    (torch.float32,  False,    "no_mask"), # calls forward_torch_softmax (apply no mask)
```

File: tests/pytorch/test_onnx_export.py [hunk 2]
```diff
@@ -1129,14 +1138,14 @@ def test_export_layernorm_mlp(
...
+    (torch.float16,  False,    "causal"),  # calls forward_torch_softmax (apply dynamic onnx mask)
+    (torch.float16,  True,     "padding"), # calls forward_torch_softmax (apply user mask)
+    (torch.float16,  False,    "padding"), # calls forward_torch_softmax (apply no mask)
+    (torch.bfloat16, False,    "causal"),  # calls forward_torch_softmax (apply dynamic onnx mask)
+    (torch.bfloat16, True,     "padding"), # calls forward_torch_softmax (apply user mask)
+    (torch.bfloat16, False,    "padding"), # calls forward_torch_softmax (apply no mask)
```
Comment (ksivaman) [inline tests/pytorch/test_onnx_export.py:1148]:
```suggestion
    (torch.float32,  True,     "padding"), # calls forward_torch_softmax (apply user mask)
    (torch.float32,  False,    "no_mask"), # calls forward_torch_softmax (apply no mask)
    (torch.float16,  False,    "causal"),  # calls forward_torch_softmax (apply dynamic onnx mask)
    (torch.float16,  True,     "padding"), # calls forward_torch_softmax (apply user mask)
    (torch.float16,  False,    "no_mask"), # calls forward_torch_softmax (apply no mask)
    (torch.bfloat16, False,    "causal"),  # calls forward_torch_softmax (apply dynamic onnx mask)
    (torch.bfloat16, True,     "padding"), # calls forward_torch_softmax (apply user mask)
    (torch.bfloat16, False,    "no_mask"), # calls forward_torch_softmax (apply no mask)
```

**Comment by ksivaman** [comment]
@nzmora-nvidia Could you sign-off your commits?

**Review by ksivaman — APPROVED** [review]
